### ISSUE TITLE
Multi-language support

### ISSUE THREAD

**Issue by geoffreymcgill** [retypeapp/retype#24]
_Originally posted by @slimooo in https://github.com/retypeapp/retype/discussions/18_

## See also

- #235 
- #18

**Comment by DannyBen** [comment]
I would like to drop the RTL bomb.... 💣

I know that RTL languages (Hebrew, Arabic) are usually neglected by most software vendors due to probably less demand, and more complexity in UI terms - but I think that if the UI is thoughtfully prepared, it should be a matter of strapping a `<body class='rtl' direction='rtl'>` to it, and then it should be easy to support RTL (which of course entails having the lefthand sidebar on the right).

Whatever the future holds for this localization feature - if RTL support finds its way in, I am happy to assist.

**Comment by fabriciomurta** [comment]
Well, I saw this bomb falling! :boom:

From my experience, this would mean a complete rewrite of the CSS styling and many HTML. As you said yourself, even right sidebars may become left sidebars when RTL is in place. So I believe translation for LTR languages should be dealt in separate steps from actual RTL implementation. That's not just a matter of placing strings over placeholders, but a complete layout revamp.

One important aspect to have it well implemented and maintained is an actual audience to report the layout issues as they rise. :)

I mean, it would be worth its own dedicated issue.

**Comment by DannyBen** [comment]
Yeah, possibly.

From my experience - and the many days I spent dealing with such issues - I want to mention just a couple of key point that might help.

1. Tables, columns, flexboxes, "float lefts" - are all automatically organized in reverse order once wrapped inside a container with `direction='rtl'` - so dealing with this is usually just minor adjustments.
2. The other aspect is CSS elements that received explicit float, align, or uneven padding/margin. For these, I usually do something like this (SCSS):

```scss
.col { 
  .ltr & { float:left; }
  .rtl & { float:right; }
}
```

Anyways - when the time comes, give me a call.

**Comment by geoffreymcgill** [comment]
RTL is certainly something I'm interested in supporting. I love the challenge. We have been involved in other large projects in the past with the requirement, so it's not completely foreign nor off our radar.

It's on my list, but... we will likely need to add to our UI team before we can tackle RTL support.

**Comment by vmohir** [comment]
This should be really easy by using CSS logical properties as Retype is using utility classes. Even there are tools to automatically convert a CSS code to logical properties. You'll lose IE support though.

**Comment by geoffreymcgill** [comment]
Retype v3.0 will include support for changing the interface language. Currently 24 languages are supported. 

See the following docs for the translations:
https://retypeapp.github.io/retype-translations/

The source repo is available at:
https://github.com/retypeapp/retype-translations

RTL is not support. Hopefully one day.

**Comment by geoffreymcgill** [comment]
The new `locale` project configuration is now available in Retype v3.0, see [docs](https://retype.com/configuration/project/#locale).

The Retype Translation repo is available at https://retypeapp.github.io/retype-translations/. 

Hope this helps.
